psiturk/psiturk_shell.py

```diff
@@ -941,60 +944,89 @@ def hit_create(self, numWorkers, reward, duration):
             print '*** duration must be greater than 0'
             return
 
-        # register with the ad server (psiturk.org/ad/register) using POST
-        if os.path.exists('templates/ad.html'):
-            ad_html = open('templates/ad.html').read()
-        else:
-            print '\n'.join(['*****************************',
-                             '  Sorry, there was an error registering ad.',
-                             '  Both ad.html is required to be in the templates folder',
-                             '  of your project so that these Ad can be served!'])
-            return
+        if use_psiturk_ad_server:
 
-        size_of_ad = sys.getsizeof(ad_html)
-        if size_of_ad >= 1048576:
-            print '\n'.join(['*****************************',
-                             '  Sorry, there was an error registering ad.',
-                             '  Your local ad.html is %s byes, but the maximum',
-                             '  template size uploadable to the Ad server is',
-                             '  1048576 bytes!' % size_of_ad])
-            return
+            # register with the ad server (psiturk.org/ad/register) using POST
+            if os.path.exists('templates/ad.html'):
+                ad_html = open('templates/ad.html').read()
+            else:
+                print '\n'.join(['*****************************',
+                                 '  Sorry, there was an error registering ad.',
+                                 '  Both ad.html is required to be in the templates folder',
```

Comment:
There's also the file size in the tastefully unique 'byes'. Worth fixing if
you're working in the file anyway, but I've avoided single figure
typo-commits because people get annoyed at them bloating the commit log.

~ John
On 23 Apr 2016 9:57 am, "Dave Eargle" <notifications@github.com> wrote:

> In psiturk/psiturk_shell.py
> <https://github.com/NYUCCL/psiTurk/pull/226#discussion_r60817102>:
>
> >
> > -        size_of_ad = sys.getsizeof(ad_html)
> > -        if size_of_ad >= 1048576:
> > -            print '\n'.join(['*****************************',
> > -                             '  Sorry, there was an error registering ad.',
> > -                             '  Your local ad.html is %s byes, but the maximum',
> > -                             '  template size uploadable to the Ad server is',
> > -                             '  1048576 bytes!' % size_of_ad])
> > -            return
> > +            # register with the ad server (psiturk.org/ad/register) using POST
> > +            if os.path.exists('templates/ad.html'):
> > +                ad_html = open('templates/ad.html').read()
> > +            else:
> > +                print '\n'.join(['*****************************',
> > +                                 '  Sorry, there was an error registering ad.',
> > +                                 '  Both ad.html is required to be in the templates folder',
>
> Yeah I had noticed that too, cringed at the grammar, then pressed on.
> You're right, this is a good time to fix it.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/NYUCCL/psiTurk/pull/226/files/6b11fc0c5017eeffd4bf279db98881ca527fc39a#r60817102>
>